tests/core/test_nlg.py

```diff
@@ -115,8 +115,25 @@ def test_nlg_fill_template_text(slot_name, slot_value):
     assert result == {"text": str(slot_value)}
 
 
+@pytest.mark.parametrize(
+    "template_text, expected",
+    [
+        ('{{"variable":"{slot_1}"}}', '{"variable":"foo"}'),
+        ("{slot_1} and {slot_2}", "foo and bar"),
+        ("{{{slot_1}, {slot_2}!}}", "{foo, bar!}"),
+    ],
+)
+def test_nlg_fill_template_text_with_json(template_text, expected):
+    template = {"text": template_text}
+    t = TemplatedNaturalLanguageGenerator(templates=dict())
+    result = t._fill_template_text(
+        template=template, filled_slots={"slot_1": "foo", "slot_2": "bar"}
+    )
+    assert result == {"text": expected}
+
+
 @pytest.mark.parametrize("slot_name, slot_value", [("tag_w_\n", "a")])
-def test_nlg_fill_template_text_w_bad_slot_name2(slot_name, slot_value):
+def test_nlg_fill_template_text_with_bad_slot_name(slot_name, slot_value):
```

Comment:
can we also add the case where there is a `{`  or `}` in a slot name?

Reply:
You raised great questions!

I'm thinking it is better to not to allow { character as part of slot name.
but if they do, it will raise ValueError 
```
>>> "{slot_one}".format(slot_one="foo")
'foo'
>>> "{slot_{one}".format(slot_one="foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: unexpected '{' in field name
```
We can let it raise here as it is or provide a friendlier wording then raise or just log as exception..... I prefer to raise otherwise you might overlook.

Also to answer the question below
>> what happens if we have {{slot_1}}?
In short, with original code, it will return '{0[slot_1]}'
```
>>> "{0[slot_1]}".format({"slot_1":"foo"})
'foo'
>>> "{{0[slot_1]}}".format({"slot_1":"foo"})
'{0[slot_1]}'
>>> "{{{0[slot_1]}}}".format({"slot_1":"foo"})
'{foo}'
```
format func will simply escape `{` thus not replacing which I expected but the only caveat is that we replace to 0[SLOT_NAME] so it looks odd...
It won't raise anything either but it should return `{slot_1}` so I updated the code.

Also rebased master